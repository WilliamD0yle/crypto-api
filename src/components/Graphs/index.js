import React, { useState } from 'react';
import moment from 'moment';
import {
    XYPlot,
    XAxis,
    YAxis,
    Hint,
    AreaSeries,
    VerticalBarSeries,
    LineMarkSeries,
    VerticalGridLines,
    HorizontalGridLines
} from 'react-vis';
import '../../../node_modules/react-vis/dist/style.css';

const axisStyle = {
    ticks: {
        fontSize: '14px',
        color: '#333'
    },
    title: {
        fontSize: '16px',
        color: '#333'
    }
};

const Graphs = props => {
    const [hoover, updateHoover] = useState(null);

    return (
        <div className="App">
            <XYPlot height={300} width={1000}>
                <VerticalGridLines />
                <HorizontalGridLines />
                <XAxis
                    title={props.xTitle}
                    tickFormat={time => `${moment.unix(time).fromNow()}`}
                    tickTotal={8}
                    style={axisStyle}
                    opacity={0.8}
                />
                <YAxis title={props.yTitle} tickFormat={value => `£${value}`} />
                {hoover ? (
                    <Hint value={hoover}>
                        <div>£{hoover.y}</div>
                    </Hint>
                ) : null}
                {props.type === 'line' ? (
                    <LineMarkSeries
                        animation={'wobbly'}
                        onValueMouseOver={value => updateHoover(value)}
                        onValueMouseOut={() => updateHoover(null)}
                        markStyle={{ stroke: 'white' }}
                        data={props.data}
                    />
                ) : props.type === 'area' ? (
                    <AreaSeries
                        animation={'wobbly'}
                        className="area-series"
                        curve="curveNatural"
                        data={props.data}
                    />
                ) : props.type === 'bar' ? (
                    <VerticalBarSeries
                        animation={'wobbly'}
                        className="bar-series"
                        data={props.data}
                    />
                ) : null}
            </XYPlot>
        </div>
    );
};

export default Graphs;
