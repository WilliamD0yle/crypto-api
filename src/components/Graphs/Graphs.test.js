import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Graphs from './';
configure({ adapter: new Adapter() });

describe('Grpah component', () => {
    it('renders line graph without crashing', () => {
        const data = [{ x: 0, y: 1 }];
        const graph = shallow(<Graphs data={data} type="line" />);
        expect(graph).toMatchSnapshot();
    });

    it('renders area graph without crashing', () => {
        const div = document.createElement('div');
        const data = [{ x: 0, y: 1 }];
        const graph = shallow(<Graphs data={data} type="area" />);
        expect(graph).toMatchSnapshot();
    });

    it('renders bar graph without crashing', () => {
        const data = [{ x: 0, y: 1 }];
        const graph = shallow(<Graphs data={data} type="bar" />);
        expect(graph).toMatchSnapshot();
    });
});
