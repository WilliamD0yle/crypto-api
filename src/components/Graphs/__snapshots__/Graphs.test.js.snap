// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grpah component renders area graph without crashing 1`] = `ShallowWrapper {}`;

exports[`Grpah component renders bar graph without crashing 1`] = `ShallowWrapper {}`;

exports[`Grpah component renders line graph without crashing 1`] = `ShallowWrapper {}`;
